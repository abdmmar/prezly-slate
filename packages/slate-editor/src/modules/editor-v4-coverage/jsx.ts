/* eslint-disable @typescript-eslint/no-namespace */

import { Coverage } from '@prezly/sdk';
import { createHyperscript } from '@prezly/slate-hyperscript';
import { COVERAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-coverage': {
                children?: ReactNode;
                coverage: {
                    id: Coverage['id'];
                };
                uuid: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-coverage': { type: COVERAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
