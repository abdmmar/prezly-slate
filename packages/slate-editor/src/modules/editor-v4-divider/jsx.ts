/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

import { DIVIDER_TYPE } from './constants';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-divider': { type: DIVIDER_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
