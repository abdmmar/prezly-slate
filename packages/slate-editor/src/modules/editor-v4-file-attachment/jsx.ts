/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { ATTACHMENT_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadcareStoragePayload } from '@prezly/uploadcare';
import type { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-file-attachment': {
                children?: ReactNode;
                description: string;
                file: UploadcareStoragePayload;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-file-attachment': { type: ATTACHMENT_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
