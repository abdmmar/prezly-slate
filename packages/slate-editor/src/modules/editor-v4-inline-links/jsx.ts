/* eslint-disable @typescript-eslint/no-namespace,@typescript-eslint/no-unused-vars */
import { LINK_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { createEditor as makeEditor } from 'slate';
import { createHyperscript, createEditor } from 'slate-hyperscript';

import { InlineLinksExtension } from './InlineLinksExtension';

import { createEditorV4 } from '#modules/editor-v4/createEditorV4';

// This allows tests to include Slate Nodes written in JSX without TypeScript complaining.
declare global {
    namespace JSX {
        interface IntrinsicElements {
            [elemName: string]: any; // eslint-disable-line
        }
    }
}

const extensions = [InlineLinksExtension()];

export const jsx = createHyperscript({
    elements: {
        link: { type: LINK_NODE_TYPE },
        paragraph: { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        editor: createEditor(() => createEditorV4(makeEditor(), () => extensions)),
    },
});
