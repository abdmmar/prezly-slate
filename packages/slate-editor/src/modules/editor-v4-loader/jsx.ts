/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

import { LOADER_TYPE } from './constants';
import { LoaderNode } from './types';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-loader': {
                children?: ReactNode;
                contentType: LoaderNode['contentType'];
                id: LoaderNode['id'];
                message: LoaderNode['message'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-loader': { type: LOADER_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
