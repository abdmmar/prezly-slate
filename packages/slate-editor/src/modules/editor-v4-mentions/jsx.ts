/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_TYPE } from '@prezly/slate-commons';
import { createHyperscript } from '@prezly/slate-hyperscript';
import { PlaceholderNode, PLACEHOLDER_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-placeholder-mention': {
                children?: ReactNode;
                key: PlaceholderNode['key'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-placeholder-mention': { type: PLACEHOLDER_NODE_TYPE },
        'h-p': { type: PARAGRAPH_TYPE },
    },
});

export default jsx;
