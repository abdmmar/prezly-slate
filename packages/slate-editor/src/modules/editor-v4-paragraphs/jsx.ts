/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
