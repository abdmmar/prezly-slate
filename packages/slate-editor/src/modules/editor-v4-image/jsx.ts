/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_TYPE } from '@prezly/slate-commons';
import { createHyperscript } from '@prezly/slate-hyperscript';
import { UploadcareStoragePayload } from '@prezly/uploadcare';
import { ReactNode } from 'react';

import { LOADER_TYPE, LoaderElementType } from '../../modules/editor-v4-loader';

import { IMAGE_CANDIDATE_TYPE, IMAGE_TYPE } from './constants';
import { ImageLayout } from './types';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-image': {
                children?: ReactNode;
                file: UploadcareStoragePayload;
                href: string;
                layout: ImageLayout;
                width: string;
                width_factor: string;
            };
            'h-image-candidate': {
                children?: ReactNode;
                href: string;
                src: string;
            };
            'h-loader': {
                children?: ReactNode;
                contentType: LoaderElementType['contentType'];
                id: LoaderElementType['id'];
                message: LoaderElementType['message'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-image': { type: IMAGE_TYPE },
        'h-image-candidate': { type: IMAGE_CANDIDATE_TYPE },
        'h-loader': { type: LOADER_TYPE },
        'h-p': { type: PARAGRAPH_TYPE },
    },
});

export default jsx;
