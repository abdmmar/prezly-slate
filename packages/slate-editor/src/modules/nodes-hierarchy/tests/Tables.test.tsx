/** @jsx jsx */

import { Alignment, ImageLayout } from '@prezly/slate-types';
import { Editor } from 'slate';

import { jsx } from '../../../jsx';

describe('nodes-hierarchy / Tables', () => {
    it('should be kept after normalization', () => {
        const editor = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                        <h:td>
                            <h:text>6</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>7</h:text>
                        </h:td>
                        <h:td>
                            <h:text>8</h:text>
                        </h:td>
                        <h:td>
                            <h:text>9</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                        <h:td>
                            <h:text>6</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>7</h:text>
                        </h:td>
                        <h:td>
                            <h:text>8</h:text>
                        </h:td>
                        <h:td>
                            <h:text>9</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toMatchSnapshot();
        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });

    it('should extract image from table cell after normalization', () => {
        const editor = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:image
                                align={Alignment.CENTER}
                                href=""
                                layout={ImageLayout.CONTAINED}
                                new_tab
                                width="100%"
                                file={{
                                    version: 2,
                                    uuid: 'c849afb1-57cc-48c2-ab31-10b5e1555a33',
                                    filename:
                                        'png-transparent-pink-cross-stroke-ink-brush-pen-red-ink-brush-ink-leave-the-material-text.png',
                                    mime_type: 'image/png',
                                    size: 41346,
                                    original_width: 920,
                                    original_height: 900,
                                    effects: [],
                                }}
                            >
                                <h:text>Caption</h:text>
                            </h:image>
                            <h:text>5</h:text>
                        </h:td>
                        <h:td>
                            <h:text>6</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>7</h:text>
                        </h:td>
                        <h:td>
                            <h:text>8</h:text>
                        </h:td>
                        <h:td>
                            <h:text>9</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:image
                    align={Alignment.CENTER}
                    href=""
                    layout={ImageLayout.CONTAINED}
                    new_tab
                    width="100%"
                    file={{
                        version: 2,
                        uuid: 'c849afb1-57cc-48c2-ab31-10b5e1555a33',
                        filename:
                            'png-transparent-pink-cross-stroke-ink-brush-pen-red-ink-brush-ink-leave-the-material-text.png',
                        mime_type: 'image/png',
                        size: 41346,
                        original_width: 920,
                        original_height: 900,
                        effects: [],
                    }}
                >
                    <h:text>Caption</h:text>
                </h:image>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                        <h:td>
                            <h:text>6</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>7</h:text>
                        </h:td>
                        <h:td>
                            <h:text>8</h:text>
                        </h:td>
                        <h:td>
                            <h:text>9</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });

    it('should remove bold mark when text is inside table cell', () => {
        const editor = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:text bold>
                                <cursor />1
                            </h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:text>
                                <cursor />1
                            </h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });

    it('should remove bold mark when text is inside paragraph inside table cell', () => {
        const editor = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:paragraph>
                                <h:text bold>
                                    <cursor />1
                                </h:text>
                            </h:paragraph>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border header={['first_row']}>
                    <h:tr>
                        <h:td>
                            <h:paragraph>
                                <h:text>1</h:text>
                            </h:paragraph>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
    });

    it('should remove empty row', () => {
        const editor = (
            <editor>
                <h:table border>
                    <h:tr></h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
    });

    it('should remove colspan', () => {
        const editor = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td colspan={2}>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
    });

    it('should restore absent cells when colspan exceeds real cells', () => {
        const editor = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td colspan={3}>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>3</h:text>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        debugger;
        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
    });

    it('should remove rowspan', () => {
        const editor = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td rowspan={2}>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td colspan={2}>
                            <h:text>2</h:text>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:table border>
                    <h:tr>
                        <h:td>
                            <h:text>1</h:text>
                        </h:td>
                        <h:td>
                            <h:text>2</h:text>
                        </h:td>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                    </h:tr>
                    <h:tr>
                        <h:td>
                            <h:paragraph>
                                <h:text></h:text>
                            </h:paragraph>
                        </h:td>
                        <h:td>
                            <h:text>4</h:text>
                        </h:td>
                        <h:td>
                            <h:text>5</h:text>
                        </h:td>
                    </h:tr>
                </h:table>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
    });
});
