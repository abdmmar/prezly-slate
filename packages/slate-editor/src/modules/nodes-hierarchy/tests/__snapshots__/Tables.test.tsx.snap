// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodes-hierarchy / Tables should be kept after normalization 1`] = `
Array [
  Object {
    "border": true,
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "1",
              },
            ],
            "type": "table-cell",
          },
          Object {
            "children": Array [
              Object {
                "text": "2",
              },
            ],
            "type": "table-cell",
          },
          Object {
            "children": Array [
              Object {
                "text": "3",
              },
            ],
            "type": "table-cell",
          },
        ],
        "type": "table-row",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "4",
              },
            ],
            "type": "table-cell",
          },
          Object {
            "children": Array [
              Object {
                "text": "5",
              },
            ],
            "type": "table-cell",
          },
          Object {
            "children": Array [
              Object {
                "text": "6",
              },
            ],
            "type": "table-cell",
          },
        ],
        "type": "table-row",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "7",
              },
            ],
            "type": "table-cell",
          },
          Object {
            "children": Array [
              Object {
                "text": "8",
              },
            ],
            "type": "table-cell",
          },
          Object {
            "children": Array [
              Object {
                "text": "9",
              },
            ],
            "type": "table-cell",
          },
        ],
        "type": "table-row",
      },
    ],
    "header": Array [
      "first_row",
    ],
    "type": "table",
  },
]
`;
